### ISSUE TITLE
Fail to forcingly terminate repair on very early stage

### ISSUE THREAD

**Issue by enaydanov** [scylladb/scylla#10384]
## Installation details

Kernel Version: 5.13.0-1021-aws

Scylla version (or git commit hash): `2022.1~rc3-20220406.5cc3b678c` with build-id `48dfae0735cd8efc4ae2f5c777beaee2a1e89f4a`

Cluster size: 4 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-9 (34.252.226.229 | 10.0.1.245) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-8 (54.75.58.163 | 10.0.0.163) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-7 (54.229.227.193 | 10.0.1.197) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-6 (18.203.126.104 | 10.0.3.203) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-5 (34.244.229.84 | 10.0.2.5) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 (34.241.60.23 | 10.0.0.83) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-3 (52.19.25.242 | 10.0.0.140) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-2 (34.254.225.189 | 10.0.3.92) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-11 (3.250.168.163 | 10.0.2.192) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-10 (54.170.171.223 | 10.0.2.120) (shards: 14)
- longevity-tls-1tb-7d-2022-1-db-node-436e2326-1 (34.250.10.44 | 10.0.2.235) (shards: 14)

OS / Image: `ami-071c70d20f0fdbb2c` (aws: eu-west-1)

Test: `longevity-1tb-7days-test`

Test id: `436e2326-9e19-4cd8-b8a8-dc9d832fe92e`

Test name: `longevity/longevity-1tb-7days-test`

Test config file(s):

- [longevity-1TB-5days-authorization-and-tls-ssl.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/7f5f150f522fd17c764c1b64e1a8f750c7a8cf18/test-cases/longevity/longevity-1TB-5days-authorization-and-tls-ssl.yaml)

## Issue description

One of runs of AbortRepair nemesis (at 2022-04-14 06:01:39) failed to abort the repair on very early stage:

```
[shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.3.92, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
[shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.2.235, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
[shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.2.5, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
[shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.0.83, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
[shard 0] repair - Aborted 1 repair job(s)  
```
But the repair continued to run.

There is also an example of a successful abort too (at 2022-04-15 04:34:06):

```
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Sending repair_flush_hints_batchlog to node=10.0.0.83, participants=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235], started
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Sending repair_flush_hints_batchlog to node=10.0.3.203, participants=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235], started
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Sending repair_flush_hints_batchlog to node=10.0.2.5, participants=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235], started
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Sending repair_flush_hints_batchlog to node=10.0.2.235, participants=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235], started
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Started to process repair_flush_hints_batchlog_request from node=10.0.2.235, target_nodes=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235], hints_timeout=300s, batchlog_timeout=300s
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Started to flush hints for repair_flush_hints_batchlog_request from node=10.0.2.235, target_nodes=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235]
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Started to flush batchlog for repair_flush_hints_batchlog_request from node=10.0.2.235, target_nodes=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235]
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Finished to flush hints for repair_flush_hints_batchlog_request from node=10.0.2.235, target_hosts=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235]
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Finished to flush batchlog for repair_flush_hints_batchlog_request from node=10.0.2.235, target_nodes=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235]
[shard 0] repair - repair[2bc1fb51-cf53-47e0-862f-ef6ab7f31343]: Finished to process repair_flush_hints_batchlog_request from node=10.0.2.235, target_nodes=[10.0.0.83, 10.0.2.5, 10.0.3.203, 10.0.2.235]
...
[shard 0] repair - Aborted 1 repair job(s)
...
[shard 0] repair - repair id [id=1, uuid=2bc1fb51-cf53-47e0-862f-ef6ab7f31343] on shard 0, keyspace=system_traces, cf=sessions_time_idx, range=(-3123832431221512806, -3103842354533520972], got error in row level repair: std::runtime_error (repair id [id=1, uuid=2bc1fb51-cf53-47e0-862f-ef6ab7f31343] is aborted on shard 0)
```

- Restore Monitor Stack command: `$ hydra investigate show-monitor 436e2326-9e19-4cd8-b8a8-dc9d832fe92e`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=436e2326-9e19-4cd8-b8a8-dc9d832fe92e)
- Show all stored logs command: `$ hydra investigate show-logs 436e2326-9e19-4cd8-b8a8-dc9d832fe92e`

## Logs:
*No logs captured during this run.*

[Jenkins job URL](https://jenkins.scylladb.com/job/enterprise-2022.1/job/longevity/job/longevity-1tb-7days-test/3/)

**Comment by enaydanov** [comment]
Note, the test run not finished yet, so, no published logs, but nodes still running.

**Comment by enaydanov** [comment]
## Logs:
- **db-cluster-436e2326.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/db-cluster-436e2326.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/db-cluster-436e2326.tar.gz)
- **monitor-set-436e2326.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/monitor-set-436e2326.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/monitor-set-436e2326.tar.gz)
- **loader-set-436e2326.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/loader-set-436e2326.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/loader-set-436e2326.tar.gz)
- **sct-runner-436e2326.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/sct-runner-436e2326.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/436e2326-9e19-4cd8-b8a8-dc9d832fe92e/20220417_174612/sct-runner-436e2326.tar.gz)

**Comment by asias** [comment]
The problem was that:

- get repair request from user, repair info is not created yet
- abort all repairs
- repair info is created and repair continues to run since the repair info is not aborted


```
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - starting user-requested repair for keyspace system_traces, repair id [id=1, uuid=db854db1-bd3c-4304-bacb-c14992a7866e], options {{ trace -> false}, { jobThreads -> 1}, { incremental -> false}, { parallelism -> parallel}, { primaryRange -> false}}
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.3.92, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.2.235, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.2.5, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Sending repair_flush_hints_batchlog to node=10.0.0.83, participants=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], started
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - Aborted 1 repair job(s)
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Started to process repair_flush_hints_batchlog_request from node=10.0.0.83, target_nodes=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83], hints_timeout=300s, batchlog_timeout=300s
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Started to flush hints for repair_flush_hints_batchlog_request from node=10.0.0.83, target_nodes=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83]
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Started to flush batchlog for repair_flush_hints_batchlog_request from node=10.0.0.83, target_nodes=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83]
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Finished to flush hints for repair_flush_hints_batchlog_request from node=10.0.0.83, target_hosts=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83]
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Finished to flush batchlog for repair_flush_hints_batchlog_request from node=10.0.0.83, target_nodes=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83]
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - repair[db854db1-bd3c-4304-bacb-c14992a7866e]: Finished to process repair_flush_hints_batchlog_request from node=10.0.0.83, target_nodes=[10.0.2.235, 10.0.2.5, 10.0.3.92, 10.0.0.83]
2022-04-14T06:01:44+00:00 longevity-tls-1tb-7d-2022-1-db-node-436e2326-4 !    INFO |  [shard 0] repair - Repair 1 out of 507 ranges, id=[id=1, uuid=db854db1-bd3c-4304-bacb-c14992a7866e], shard=0, keyspace=system_traces, table={node_slow_log, events, sessions, node_slow_log_time_idx, sessions_time_idx}, range=(-9218802963350676433, -9217166265207825724], peers={10.0.3.92}, live_peers={10.0.3.92}
```

As a workaround, one can run abort request again. 

@enaydanov In your test when did you run abort repair? Did you wait for the start repair request to finish before you start the abort one after another or in parallel? Can you show me the relevant test code?

The issue exists especially when we flush hints before repair. This is a low priority issue with low impact on user. The race window (get repair request and creates repair info) is small comparing to whole repair time.

**Comment by enaydanov** [comment]
Yes, we're waiting while number of active repair changed from 0 to 1 using `http://127.0.0.1:10000/storage_service/active_repair/` API call: https://github.com/scylladb/scylla-cluster-tests/blob/fa6ed9c2efde0b25bde7299889b4284a1f15f4a7/sdcm/nemesis.py#L2108

The test flow: https://github.com/scylladb/scylla-cluster-tests/blob/fa6ed9c2efde0b25bde7299889b4284a1f15f4a7/sdcm/nemesis.py#L2119

**Comment by hansh0801** [comment]
I have same issue too.

**Comment by slivne** [comment]
@asias 

> The issue exists especially when we flush hints before repair. This is a low priority issue with low impact on user. The race window (get repair request and creates repair info) is small comparing to whole repair time.

repair was created it does exist as far as the user is concerned `repair id [id=1, uuid=db854db1-bd3c-4304-bacb-c14992a7866e]`

I disagree with this - we need to make abort work - not have the user check if abort worked and then attempt to abort again (and again ...).

I don;t think we should block the release on this - yet we need to fix this - we can list it as an open issue for 5.0.0

**Comment by asias** [comment]
> @asias
> 
> > The issue exists especially when we flush hints before repair. This is a low priority issue with low impact on user. The race window (get repair request and creates repair info) is small comparing to whole repair time.
> 
> repair was created it does exist as far as the user is concerned `repair id [id=1, uuid=db854db1-bd3c-4304-bacb-c14992a7866e]`
> 
> I disagree with this - we need to make abort work - not have the user check if abort worked and then attempt to abort again (and again ...).

I am not sure why are disagreeing with the workaround. 

```
As a workaround, one can run abort request again.
```

Of course, check again and again is not a solution. 

> I don;t think we should block the release on this - yet we need to fix this - we can list it as an open issue for 5.0.0

Yes.

**Comment by asias** [comment]
PR is sent: https://github.com/scylladb/scylla/pull/10428

**Comment by avikivity** [comment]
@enaydanov was this a regression? I'm evaluating whether to backport or not.

I'm leaning not to, it looks like a minor problem.

**Comment by enaydanov** [comment]
I didn't find any evidence of this problem in the past, so, it looks like a regression, but the problem is pretty minor for me.

**Comment by mykaul** [comment]
> I didn't find any evidence of this problem in the past, so, it looks like a regression, but the problem is pretty minor for me.

Based on the above, removing the backport candidate label.
